src/test/protocol/KnownFormatToGRPC_test.cpp

```diff
@@ -0,0 +1,924 @@
+                //
+                // Some of these may be bugs.
+                static const std::
+                    map<LedgerEntryType, std::vector<SField const*>>
+                        gRPCOmitFields{
+                            {ltACCOUNT_ROOT, {&sfWalletLocator, &sfWalletSize}},
+                            {ltDIR_NODE, {&sfExchangeRate}},
+                            {ltLEDGER_HASHES, {&sfFirstLedgerSequence}},
+                            {ltPAYCHAN, {&sfDestinationNode}},
+                        };
+
+                if (auto const iter = gRPCOmitFields.find(fmtId);
+                    iter != gRPCOmitFields.end())
+                {
+                    std::vector<SField const*> const& omits = iter->second;
+
+                    // Check for fields that gRPC omits from this type.
+                    if (std::find_if(
+                            omits.begin(),
+                            omits.end(),
+                            [&sField](SField const* const omit) {
+                                return *omit == sField;
+                            }) != omits.end())
+                    {
+                        // This is one of the fields that gRPC omits.
+                        continue;
+                    }
+                }
+            }
+
+            // The SFields and gRPC disagree on the names of some fields.
+            // Provide a mapping from SField names to gRPC names for the
+            // known exceptions.
+            //
+            // At least one gRPC spelling is a bug.
+            // clang-format off
+            static const std::map<std::string, std::string> sFieldToGRPC{
+                {"AccountTxnID",      "AccountTransactionID"},
+                {"Fee",               "XRPDropsAmount"},
+                {"PreviousTxnID",     "PreviousTransactionID"},
+                {"PreviousTxnLgrSeq", "PreviousTransactionLedgerSequence"},
+                {"Signature",         "PaymentChannelSignature"},
+                {"SigningPubKey",     "SigningPublicKey"},
+                {"TakerGetsCurrency", "TakerGetsCurreny"},
+                {"TxnSignature",      "TransactionSignature"},
+            };
+            // clang-format on
+
+            std::string gRPCName = sFieldToGRPC.count(sField.getName()) == 1
+                ? sFieldToGRPC.at(sField.getName())
+                : sField.getName();
+
+            sFields.insert({std::move(gRPCName), &sField});
+        }
+        return sFields;
+    }
+
+    // Given a Descriptor for a KnownFormat and a map of the SFields of that
+    // KnownFormat, make sure the fields are aligned.
+    void
+    validateDescriptorAgainstSFields(
+        google::protobuf::Descriptor const* const pbufDescriptor,
+        std::optional<google::protobuf::Descriptor const* const> commonFields,
+        std::string const& knownFormatName,
+        std::map<std::string, SField const*> sFields)
```

Comment:
What do you think about making `sFields` an rvalue ref? I know we often make parameters like this value-types, so no change needed.